packages/ember/tests/routing/substates_test.js

```diff
@@ -267,6 +267,38 @@ moduleFor(
       });
     }
 
+    ['@test Bad timing issue ☠️'](assert) {
+      let appDeferred = RSVP.defer();
+
+      this.router.map(function() {
+        this.route('dummy');
+      });
+
+      this.addTemplate('application_loading', `
+        <div id="toplevel-loading">TOPLEVEL LOADING</div>
+      `);
+      this.add('route:application', Route.extend({
+        model() {
+          return appDeferred.promise;
+        }
+      }));
+
+      this.add('route:dummy', Route.extend({
+        redirect() {
+          return RSVP.resolve()
```

Comment:
If I understand well, this is specifically what the test is about ?

Reply:
Correct.